src/ex1/scripts/domScripts.js

```diff
@@ -0,0 +1,195 @@
+const taskInput = document.querySelector(".task-input");
+const addTaskButton = document.querySelector(".task-add-btn");
+const tasksList = document.querySelector(".tasks-list");
+const filterOptiopn = document.querySelector(".filter-tasks");
+const searchInput = document.querySelector(".search-input");
+
+addTaskButton.addEventListener("click", onAddTaskClick);
+tasksList.addEventListener("click", onTaskButtonsClick);
+filterOptiopn.addEventListener("click", onFilterOptionChange);
+searchInput.addEventListener("keyup", onSearchInputKeyUp);
+
+taskInput.addEventListener("keypress", function (event) {
+  if (event.key === "Enter") {
+    event.preventDefault();
+    addTaskButton.click();
+  }
+});
+
+function onAddTaskClick(e) {
+  const isOk = canProceed();
+  if (isOk) {
+    const taskDiv = document.createElement("div");
+    taskDiv.classList.add("task");
+
+    const gripLines = createGridLines();
+    taskDiv.appendChild(gripLines);
+
+    const taskContent = createTaskContent();
+    taskDiv.appendChild(taskContent);
+
+    const completeTaskButton = createCompleteBtn();
+    taskDiv.appendChild(completeTaskButton);
+
+    const removeTaskButton = createRemoveBtn();
+    taskDiv.appendChild(removeTaskButton);
+
+    addTaskDivEventListeners(taskDiv);
+    tasksList.appendChild(taskDiv);
+    saveTaskToLocalStorage(taskContent.innerText, false);
+    toggleEmptyMsg();
+    taskInput.value = "";
+  }
+}
+
+function canProceed() {
+  const isAlertOn = isAlertShown();
```

Comment:
canProceed suppose to be a boolean function but also does other functionality which is not a good practice.
when created validate / boolean functions it is better to focus only on the conditional logic and not to do other stuff (because logic wise you do not expect something to happen when you ask a question like "canProceed")

here you could just do:
`function canProceed() {
   return taskInput.value.**trim()** != ""
}`